### PR TITLE
N epoch test

### LINKED ISSUE #626: [NEW FEATURE] Key check for multi-epoch input
PSF interpolation (PSFEx and MCCD) in multi-epoch mode requires the key `N_EPOCH` key / column in the input
(SExtractor) catalogue. If the previous module was not run in multi-epoch mode, this key is not present.
At the moment there is no check, and the code returns with an error.

A better solution is to raise an error.

### PR DESCRIPTION
**Pull request by martinkilbinger**
## Summary

In the PSFEx and MCCD interpolation modules, a check is added for the input key 'N_EPOCH'. A KeyError is raised if not present. Closes #626 .

## Reviewer Checklist

> Reviewers should tick the following boxes before approving and merging the PR.

- [x] The PR targets the `develop` branch
- [x] The PR is assigned to the developer
- [x] The PR has appropriate labels
- [x] The PR is included in appropriate projects and/or milestones
- [x] The PR includes a clear description of the proposed changes
- [x] If the PR addresses an open issue the description includes "closes #<ISSUE NUMBER>"
- [x] The code and documentation style match the current standards
- [x] Documentation has been added/updated consistently with the code
- [ ] All CI tests are passing
- [x] API docs have been built and checked at least once (if relevant)
- [x] All changed files have been checked and comments provided to the developer
- [ ] All of the reviewer's comments have been satisfactorily addressed by the developer

### CODE REVIEW

**Comment by martinkilbinger** [comment]
@lbaumo when you get a chance you can check this PR and accept (if everything is ok). The CI tests are failing due to the pysap issue, but we can ignore this here since only the documentation changed.

File: shapepipe/modules/mccd_package/mccd_interpolation_script.py [hunk 1]
```diff
@@ -389,7 +394,13 @@ def _interpolate_me(self):
         cat.open()
 
         all_id = np.copy(cat.get_data()['NUMBER'])
-        n_epoch = np.copy(cat.get_data()['N_EPOCH'])
+        key_ne = 'N_EPOCH'
+        if key_ne not in cat.get_data():
+            raise KeyError(
+                f'Key {key_ne} not found; run previous module (SExtractor)'
```
Comment (sfarrens) [inline shapepipe/modules/mccd_package/mccd_interpolation_script.py]: Perhaps you can specify here where the code is looking for this key.

File: shapepipe/modules/psfex_interp_package/psfex_interp.py [hunk 2]
```diff
@@ -562,7 +567,13 @@ def _interpolate_me(self):
         cat.open()
 
         all_id = np.copy(cat.get_data()['NUMBER'])
-        n_epoch = np.copy(cat.get_data()['N_EPOCH'])
+        key_ne = 'N_EPOCH'
+        if key_ne not in cat.get_data():
+            raise KeyError(
+                f'Key {key_ne} not found; run previous module (SExtractor)'
```
Comment (sfarrens) [inline shapepipe/modules/psfex_interp_package/psfex_interp.py]: Same here.

**Review by sfarrens — CHANGES_REQUESTED** [review]

File: shapepipe/modules/mccd_package/mccd_interpolation_script.py (hunk 1 above)
Comment (martinkilbinger) [reply shapepipe/modules/mccd_package/mccd_interpolation_script.py]: Done

File: shapepipe/modules/psfex_interp_package/psfex_interp.py (hunk 2 above)
Comment (martinkilbinger) [reply shapepipe/modules/psfex_interp_package/psfex_interp.py]: Done

**Review by sfarrens — APPROVED** [review]

**Pull request merged**